client/state/site-settings/selectors.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * Returns true if we are requesting settings for the specified site ID, false otherwise.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Number}  siteId Site ID
+ * @return {Boolean}        Whether site settings is being requested
+ */
+export function isRequestingSiteSettings( state, siteId ) {
+	return get( state.siteSettings.requesting, [ siteId ], false );
+}
+
+/**
+ * Returns true if we are saving settings for the specified site ID, false otherwise.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Number}  siteId Site ID
+ * @return {Boolean}        Whether site settings is being requested
+ */
+export function isSavingSiteSettings( state, siteId ) {
```

Comment:
Why do you think we'll need this (and the related reducer)? Do you worry that if used, this could prevent multiple concurrent requests to update _different_ site settings, which would be perfectly valid?